apps/settings/js/modules/mvvm/observable.js

```diff
@@ -1,102 +1,219 @@
+define(function(require) {
+  'use strict';
+
+  var DependencyMap = require('modules/base/dependency_map');
+
+  // Observable
+  var OP_PREFIX = (name) => { return '$OP_' + name; };
+
+  function Observable(obj) {
+    this._init(obj);
+  }
+
+  Observable.prototype._init = function o_init(obj) {
+    if (!obj) {
+      return;
+    }
+    for (var name in obj) {
+      // If name is a function, simply add it to the observable.
+      if (typeof obj[name] === 'function') {
+        this[name] = obj[name];
+      } else {
+        _defineObservableProperty(this, name, {
+          value: obj[name]
+        });
+      }
+    }
+  };
+
+  Observable.prototype._notify = function o__notify(name, newValue, oldValue) {
+    var observers = !!this._observers && this._observers[name];
+    if (observers) {
+      observers.forEach(function(observer) {
+        observer(newValue, oldValue);
+      });
+    }
+  };
+
+  Observable.prototype._removeObserver =
+    function o__removeObserver(observer, name) {
+      // arguments in reverse order to support .bind(observer) for the
+      // unbind from all case
+      var observers = !!this._observers && this._observers[name];
+      if (!observers) {
         return;
       }
-      var index = handlers.indexOf(handler);
+      var index = observers.indexOf(observer);
       if (index >= 0) {
-        handlers.splice(index, 1);
+        observers.splice(index, 1);
       }
+  };
+
+  Observable.prototype.observe = function o_observe(name, observer) {
+    var propertyInfo =
+      this._observablePropertiesInfo && this._observablePropertiesInfo[name];
+    if (!propertyInfo) {
+      throw new Error('Observable: property ' + name + ' does not exist');
+    }
+
+    if (typeof observer !== 'function') {
+      return;
+    }
+
+    if (!this._observers) {
+      this._observers = {};
+    }
+    var observers = this._observers[name];
+    if (typeof observers === 'undefined') {
+      observers = this._observers[name] = [];
     }
+    observers.push(observer);
+  };
 
-    if (typeof prop === 'function') {
-      // (handler) -- remove from every key in _eventHandlers
-      Object.keys(_eventHandlers).forEach(removeHandler.bind(null, prop));
-    } else if (handler) {
-      // (prop, handler) -- remove handler from the specific prop
-      removeHandler(handler, prop);
-    } else if (prop in _eventHandlers) {
-      // (prop) -- otherwise remove all handlers for property
-      _eventHandlers[prop] = [];
+  Observable.prototype.unobserve = function o_unobserve(name, observer) {
+    if (typeof name === 'function') {
+      // (observer) -- remove from every key in _observers
+      Object.keys(this._observers).forEach(
+        this._removeObserver.bind(this, name));
+    } else {
+      var propertyInfo =
+      this._observablePropertiesInfo && this._observablePropertiesInfo[name];
+      if (!propertyInfo) {
+        throw new Error('Observable: property ' + name + ' does not exist');
```

Comment:
Same question here, maybe we can just console and return directly ? This may happen sometimes if developers don't notice this and I think this is not a big deal and can still make the program work.